lib/Swagger/src/Discussion/Api/LegacyRedirectsApi.php

```diff
@@ -0,0 +1,317 @@
+            );
+        }
+        // path params
+        if ($board_id !== null) {
+            $resourcePath = str_replace(
+                "{" . "boardId" . "}",
+                $this->apiClient->getSerializer()->toPathValue($board_id),
+                $resourcePath
+            );
+        }
+        
+        // for model (json/xml)
+        if (isset($_tempBody)) {
+            $httpBody = $_tempBody; // $_tempBody is the method argument, if present
+        } elseif (count($formParams) > 0) {
+            $httpBody = $formParams; // for HTTP post (form)
+        }
+        // this endpoint requires API key authentication
+        $apiKey = $this->apiClient->getApiKeyWithPrefix('X-Wikia-AccessToken');
+        if (strlen($apiKey) !== 0) {
+            $headerParams['X-Wikia-AccessToken'] = $apiKey;
+        }
+        // this endpoint requires API key authentication
+        $apiKey = $this->apiClient->getApiKeyWithPrefix('X-Wikia-UserId');
+        if (strlen($apiKey) !== 0) {
+            $headerParams['X-Wikia-UserId'] = $apiKey;
+        }
+        // make the API Call
+        try {
+            list($response, $statusCode, $httpHeader) = $this->apiClient->callApi(
+                $resourcePath,
+                'GET',
+                $queryParams,
+                $httpBody,
+                $headerParams,
+                '\Swagger\Client\Discussion\Models\LegacyRedirectResponse',
+                '/{siteId}/redirect/board/{boardId}'
+            );
+
+            return [$this->apiClient->getSerializer()->deserialize($response, '\Swagger\Client\Discussion\Models\LegacyRedirectResponse', $httpHeader), $statusCode, $httpHeader];
+        } catch (ApiException $e) {
+            switch ($e->getCode()) {
+                case 200:
+                    $data = $this->apiClient->getSerializer()->deserialize($e->getResponseBody(),
+	                    '\Swagger\Client\Discussion\Models\LegacyRedirectResponse',
+	                    $e->getResponseHeaders());
+                    $e->setResponseObject($data);
+                    break;
+                case 404:
+                    $data = $this->apiClient->getSerializer()->deserialize($e->getResponseBody(),
+	                    '\Swagger\Client\Discussion\Models\HalProblem', $e->getResponseHeaders());
+                    $e->setResponseObject($data);
+                    break;
+            }
+
+            throw $e;
+        }
+    }
+
+    /**
+     * Operation getThreadRedirect
+     *
+     * Legacy Thread to Discussion Thread
+     *
+     * @param int $site_id The id of the site (required)
+     * @param int $thread_id Legacy Forum thread ID (required)
+     * @throws \Swagger\Client\ApiException on non-2xx response
+     * @return \Swagger\Client\Discussion\Models\LegacyRedirectResponse
+     */
+    public function getThreadRedirect($site_id, $thread_id)
+    {
+        list($response) = $this->getThreadRedirectWithHttpInfo($site_id, $thread_id);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "$response". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)